src/main/scala/uk/gov/nationalarchives/aggregate/processing/modules/AssetProcessing.scala

```diff
@@ -0,0 +1,64 @@
+package uk.gov.nationalarchives.aggregate.processing.modules
+
+import com.typesafe.config.{Config, ConfigFactory}
+import com.typesafe.scalalogging.Logger
+import uk.gov.nationalarchives.aggregate.processing.modules.AssetProcessing.{AssetProcessingError, AssetProcessingResult, logger}
+import uk.gov.nationalarchives.aggregate.processing.modules.ErrorHandling.BaseError
+import uk.gov.nationalarchives.aggregate.processing.utilities.UTF8ValidationHandler
+import uk.gov.nationalarchives.aws.utils.s3.{S3Clients, S3Utils}
+import uk.gov.nationalarchives.utf8.validator.Utf8Validator
+
+import scala.util.{Failure, Try}
+
+class AssetProcessing(s3Utils: S3Utils)(implicit logger: Logger) {
+  //  Class to handle the processing of the metadata JSON sidecar
+  def processAsset(s3Bucket: String, objectKey: String): Map[String, AssetProcessingResult] = {
+    logger.info(s"Processing asset metadata for: $objectKey")
+    val inputStream = s3Utils.getObjectAsStream(s3Bucket, objectKey)
+    validUTF8(inputStream, s3Bucket, objectKey)
+    /* TODO:
+     *  - convert to Json object
+     *  - check if contains and TDR 'supplied' fields, if so create 'SuppliedMetadata' case classes for each value
+     *  - create 'ClientSideMetadataInput' Consignment API case class
+     *  - write error Json to S3 if appropriate
+     *  - return Map of matchId -> case class ('ClientSideMetadataInput', List[SuppliedMetadata], ProcessingErrors)
+     */
+
+    Map()
+  }
+
+  private def validUTF8(inputStream: java.io.InputStream, s3Bucket: String, objectKey: String): Unit = {
+    val utf8Validator = new Utf8Validator(new UTF8ValidationHandler()(logger))
+    Try(utf8Validator.validate(inputStream)) match {
+      case Failure(ex) =>
+        val consignmentId = objectKey.split('/')(2)
+        val matchId = objectKey.split('/').last
```

Comment:
I think this provide: `matchId.metadata` . Just want the "matchId" value, without the `.metadata` suffix.